src/app/home/components/Keystore.tsx

```diff
@@ -0,0 +1,158 @@
+import React from "react";
+import { Block, BlockTypes } from "@/components/Block";
+import { Button } from "@/components/Button";
+import { Subtitle } from "@/components/Subtitle";
+import { useRLN, useStore, useWallet } from "@/hooks";
+import { useKeystore } from "@/hooks/useKeystore";
+
+export const Keystore: React.FunctionComponent<{}> = () => {
+  const { keystoreCredentials } = useStore();
+  const { onGenerateCredentials } = useWallet();
+  const { onReadCredentials, onRegisterCredentials } = useKeystore();
+
+  const { password, onPasswordChanged } = usePassword();
+  const { selectedKeystore, onKeystoreChanged } = useSelectedKeystore();
+  const { onExportKeystore, onImportKeystoreFileChange } =
+    useImportExportKeystore();
+
+  const credentialsNodes = React.useMemo(
+    () =>
+      keystoreCredentials.map((v) => (
+        <option key={v} value={v}>
+          {v}
+        </option>
+      )),
+    [keystoreCredentials]
+  );
+
+  return (
+    <Block className="mt-10">
+      <Block type={BlockTypes.FlexHorizontal}>
+        <Subtitle>Keystore</Subtitle>
+        <div>
+          <Button>
+            <label htmlFor="keystore-import" className="cursor-pointer">
+              Import
+            </label>
+          </Button>
+          <input
+            id="keystore-import"
+            type="file"
+            className="hidden"
+            onChange={onImportKeystoreFileChange}
+          />
+          <Button className="ml-2" onClick={onExportKeystore}>
+            Export
+          </Button>
+        </div>
+      </Block>
+
+      <Block className="mt-4">
+        <label
+          htmlFor="keystore-input"
+          className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
+        >
+          Password(used for reading/saving into Keystore)
+        </label>
+        <input
+          type="text"
+          value={password}
+          id="keystore-input"
+          onChange={onPasswordChanged}
+          className="bg-gray-50 border border-gray-300 text-gray-900 text-sm w-full rounded-lg focus:ring-blue-500 focus:border-blue-500 block p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
+        />
+      </Block>
+
+      <Block className="mt-4">
+        <p className="text-s mb-2">Generate new credentials from wallet</p>
```

Comment:
can we merge "Generate new credentials" and "Register credentials" into a single "Register credentials"? that does both?

afaik registering credentials imply also to generate them before. So with this we can save the user a click.

Reply:
https://github.com/waku-org/waku-frontend/pull/10